3ds2/src/main/java/com/adyen/checkout/adyen3ds2/internal/data/api/SubmitFingerprintRepository.kt

```diff
@@ -50,7 +50,7 @@ internal class SubmitFingerprintRepository internal constructor(
             }
             else -> {
                 Logger.e(TAG, "submitFingerprint: unexpected response $response")
-                throw IllegalStateException("Failed to retrieve 3DS2 fingerprint result")
+                error("Failed to retrieve 3DS2 fingerprint result")
```

Comment:
Is detekt complaining about `IllegalStateException`s?

Reply:
I like the `error()` and `check()` functions, I was just asking out of curiosity.